frontend/include/chpl/types/Type.h

```diff
@@ -230,8 +230,8 @@ class Type {
   }
 
   /** returns true for a type that is a kind of pointer */
-  bool isPtrType() const {
-    return isClassType() || isCFnPtrType() || isCVoidPtrType() || isCPtrType();
+  bool isAnyPtrType() const {
```

Comment:
Maybe we have to rename it (since `isPtrType` will now check if it's a subclass of `PtrType`). But, `isAny___` normally checks for "is it the generic type that represents some class of things?" As in, we have `AnyImagType` as a generic type that means any `imag` type of any width. So `isAnyImagType` checks if it is that thing.

I can think of lots of ways out of this situation, but they are all perhaps more of a side trip than I expect you or I want for this PR. E.g., we could rename all of the `Any___` types to `GenericTypeAny___` or something. If you want to do that, it's fine with me, but needs to be a separate PR. 

Calling it `isPointerType` or `isPointerLike` or `isPtrishType` would be the short way.